tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -944,33 +950,54 @@ def _create_dummy_array_for_dot(xp, shape, dtype):
 ]
 
 
-@chainerx.testing.numpy_chainerx_array_equal()
-@pytest.mark.parametrize_device(['native:0', 'cuda:0'])
-@pytest.mark.parametrize('input', [
-    numpy.asarray(0), numpy.asarray(-4), numpy.asarray(4),
-    numpy.asarray(-float('inf')), numpy.asarray(float('inf')),
-    numpy.asarray(float('nan')),
-    numpy.full((), 2), numpy.full((0,), 2), numpy.full((2, 3), 2)
-])
-@pytest.mark.parametrize('in_dtype,out_dtype', _expected_dtypes_math_functions)
-def test_exp(xp, device, input, in_dtype, out_dtype):
-    a = xp.array(input.astype(in_dtype))
-    a = dtype_utils.cast_if_numpy_array(xp, a, out_dtype)
-    return xp.exp(a)
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'shape': [(), (0,), (1,), (2, 0, 3), (1, 1, 1), (2, 3)],
+        'expected_dtypes': _expected_dtypes_math_functions,
+        'input': [0, 2, -2],
+        'check_numpy_strides_compliance': [False],
```

Comment:
Would you mind leaving a comment (either here or in code) why this check must be skipped?

Reply:
`dtype_utils.cast_if_numpy_array` sometimes changes the shape of given numpy arrays if the size is equal to 0.

Reply:
In that case, could you condition the flag based on the size dynamically (or create a slightly more complicated parameterization...)?

Reply:
Ah, you fixed it. Thanks!